java-checks-test-sources/default/src/main/java/checks/JpaEagerFetchTypeCheckSample.java

```diff
@@ -0,0 +1,64 @@
+package checks;
+
+import jakarta.persistence.Basic;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.OneToOne;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+import java.util.List;
+import static jakarta.persistence.FetchType.EAGER;
+
+public class JpaEagerFetchTypeCheckSample {
+  private enum CustomFetchType {
+    EAGER, LAZY
+  }
+
+  @Target(ElementType.FIELD)
+  private @interface CustomAnnotation {
+    FetchType value() default FetchType.LAZY;
+    CustomFetchType fetch() default CustomFetchType.LAZY;
+  }
+
+  @Basic(fetch = FetchType.EAGER) // Noncompliant [[sc=28;ec=33]] {{Use lazy fetching instead.}}
+  private String foo;
+
+  @javax.persistence.Basic(fetch = javax.persistence.FetchType.EAGER) // Noncompliant
+  private String foo2;
+
+  @Basic(optional = true, fetch = FetchType.EAGER) // Noncompliant
+  private String baz;
+
+  @Basic(fetch = EAGER) // Noncompliant
+  private String baz4;
+
+  @OneToMany(fetch = FetchType.EAGER) // Noncompliant
+  private List<String> foos;
+
+  @javax.persistence.OneToMany(fetch = javax.persistence.FetchType.EAGER) // Noncompliant
+  private List<String> foos2;
+
+  @Basic(fetch = FetchType.LAZY) // Compliant
+  private String bar;
+
+  @javax.persistence.Basic(fetch = javax.persistence.FetchType.LAZY) // Compliant
+  private String bar2;
+
+  @OneToMany(fetch = FetchType.LAZY) // Compliant
+  private List<String> bars;
+
+  @javax.persistence.OneToMany(fetch = javax.persistence.FetchType.LAZY) // Compliant
+  private List<String> bars2;
+
+  @Basic // Compliant
+  private String baz2;
+
+  @OneToOne // Compliant
+  private String baz3;
+
+  @CustomAnnotation(fetch = CustomFetchType.EAGER) // Compliant
+  private String custom;
+
+  @CustomAnnotation(EAGER) // Compliant
+  private String customNoAssignment;
+}
```

Comment:
If we add two more test cases, this reveals an inconsistency: either we let the rule report also on custom annotations. Then the case can happen that a `FetchType` argument is not an assignment and line 62 in your samples should be `// Noncompliant` instead of `// Compliant`. 

_OR:_ we might choose to ignore custom annotations. Then sample line 62 is OK, but my added sample `@CustomAnnotationSwapped(fetch = EAGER) // Noncompliant` should then also be compliant instead.

Since we do not want to manage a list of all non-custom annotations that take a `FetchType` as an argument, I would suggest the implementation should not check for assignment arguments only.

```Java
  @Target(ElementType.FIELD)
  private @interface CustomAnnotationSwapped {
    CustomFetchType value() default CustomFetchType.LAZY;
    FetchType fetch() default FetchType.LAZY;
  }
  
  @CustomAnnotationSwapped(fetch = EAGER) // Noncompliant
  private String customSwapped;

  @CustomAnnotationSwapped(CustomFetchType.EAGER) // Compliant
  private String customNoAssignmentSwapped;
```